trestle/lib/osco.py

```diff
@@ -0,0 +1,297 @@
+        """Extract the rules from the input data."""
+        while True:
+            resource = self._content
+            if resource is None:
+                logger.debug(f'resource: {resource}')
+                break
+            if 'kind' not in resource.keys():
+                logger.debug('kind: not found')
+                break
+            if resource['kind'] != 'ConfigMap':
+                logger.debug(f'kind: {resource["kind"]}')
+                break
+            if 'results' not in resource['data']:
+                logger.debug('results: not found')
+                break
+            if 'metadata' not in resource.keys():
+                logger.debug('metadata: not found')
+                self._metadata['name'] = None
+                self._metadata['namespace'] = None
+            else:
+                self._metadata['name'] = resource['metadata']['name']
+                self._metadata['namespace'] = resource['metadata']['namespace']
+            results = resource['data']['results']
+            if results.startswith('<?xml'):
+                pass
+            else:
+                results = bz2.decompress(base64.b64decode(results))
+            logger.debug('========== <results> ==========')
+            logger.debug(results)
+            logger.debug('========== </results> ==========')
+            self._parse_xml(results, self._metadata['name'])
+            break
+
+    def _parse_xml(self, results, name):
+        """Parse the stringified XML."""
+        root = ElementTree.fromstring(results)
+        target = self._get_target(root)
+        config_map = self._get_config_map(root)
+        stats = {}
+        for lev1 in root:
+            tag = _remove_namespace(lev1.tag)
+            if tag == 'rule-result':
+                idref = lev1.get('idref')
+                time = lev1.get('time')
+                result = self._get_result(lev1)
+                self._post(config_map, idref, time, result, target, name)
+                if result not in stats.keys():
+                    stats[result] = 0
+                stats[result] += 1
+            elif tag == 'benchmark':
+                self._benchmark['href'] = lev1.get('href')
+                self._benchmark['id'] = lev1.get('id')
+
+    def _get_target(self, root):
+        """Extract 'target' info from the stringified XML."""
+        target = None
+        for lev1 in root:
+            tag = _remove_namespace(lev1.tag)
+            if tag == 'target':
+                target = root.find(lev1.tag).text
+                break
+        return target
+
+    def _get_config_map(self, root):
+        """Extract 'config_map' info from the stringified XML."""
+        retval = ''
+        for lev1 in root:
+            tag = _remove_namespace(lev1.tag)
+            if tag == 'benchmark':
+                href = lev1.get('href')
+                retval = href.split('.')[0].rsplit('/', 1)[1]
+                break
+        return retval
```

Comment:
Is this a config map or something else? I think this is the benchmark ID we are removing here.